src/hooks/sdk/useAttendeeVideoMetrics.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState, useEffect } from 'react';
+import { ClientMetricReport } from 'amazon-chime-sdk-js';
+
+import { useAudioVideo } from '../../providers/AudioVideoProvider';
+
+interface AttendeeVideoMetrics { 
+  streams: { [id: string]: {[key: string]: number} };
+};
+
+export function useAttendeeVideoMetrics(attendeeId: string) {
+  const audioVideo = useAudioVideo();
+  const [metrics, setMetrics] = useState<AttendeeVideoMetrics>({
+    streams: {},
+  });
+
+  useEffect(() => {
+    if (!audioVideo) {
+      return;
+    }
+
+    const observer = {
+      metricsDidReceive(clientMetricReport: ClientMetricReport): void {
+        const videoMetricReport = clientMetricReport.getObservableVideoMetrics();
+        setMetrics({
+          streams: videoMetricReport[attendeeId],
+        });
+      },
+    }
+
+    audioVideo.addObserver(observer);
+
+    return () => audioVideo.removeObserver(observer);
+  }, [audioVideo]);
```

Comment:
Should this also have a dependency on `attendeeId`? So, when the passed `attendeeId` change re-run this `useEffect`?

Reply:
Do you mean when attendeeId changes, run this removeObserver in useEffect?

Reply:
Hmm, the `useEffect` itself not only the `removeObserver`. I think we don't need the `attendeeId` dependency here since `metricsDidReceive` does not depend on the passed `attendeeId`. The `videoMetricReport` will have all the attendeeIds right? So, no change required here. I was thinking whether the `attendeeId` update affects this `useEffect` and if this `useEffect` needs to re-run.